File: lib/limelight/players/check_box.rb
#- Copyright 2008 8th Light, Inc.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

module Limelight
  module Players
    
    module CheckBox
      class << self
        def extended(prop)
          check_box = Limelight::UI::Model::Inputs::CheckBoxPanel.new
          prop.panel.add(check_box)
          prop.check_box = check_box.check_box
        end
      end
      
      attr_accessor :check_box
      
      def checked=(value)
        check_box.selected = value
      end
      
      def checked
        return check_box.selected?
      end
      alias :checked? :checked
  
    end
  
  end
end

File: src/limelight/ui/model/inputs/CheckBox.java
//- Copyright 2008 8th Light, Inc. All Rights Reserved.
//- Limelight and all included source files are distributed under terms of the GNU LGPL.

package limelight.ui.model.inputs;

import limelight.ui.model.updates.Updates;
import limelight.ui.model.updates.BoundedPaintUpdate;

import javax.swing.*;
import java.awt.*;

public class CheckBox extends JCheckBox
{
  private CheckBoxPanel panel;

  public CheckBox(CheckBoxPanel panel)
  {
    this.panel = panel;
  }

  public void repaint()
  {
    if(panel != null)
      panel.setNeededUpdate(Updates.paintUpdate);
  }

  public void repaint(long tm, int x, int y, int width, int height)
  {
    if(panel != null)
      panel.setNeededUpdate(new BoundedPaintUpdate(x, y, width, height));
  }

  public void repaint(Rectangle r)
  {
    if(panel != null)
      panel.setNeededUpdate(new BoundedPaintUpdate(r));
  }

  public boolean isShowing()
  {
    return true;
  }
}
